diff --git a/fsst/fsst_avx512.cpp b/fsst/fsst_avx512.cpp
index a2b7b5e5..1b974f95 100644
--- a/fsst/fsst_avx512.cpp
+++ b/fsst/fsst_avx512.cpp
@@ -20,7 +20,7 @@
 #if defined(__x86_64__) || defined(_M_X64)
 #include <immintrin.h>
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 bool fsst_hasAVX512() {
    int info[4];
    __cpuidex(info, 0x00000007, 0);
@@ -43,11 +43,11 @@ bool fsst_hasAVX512() { return false; }
 // In one call of this function, we can compress 512 strings, each of maximum length 511 bytes.
 // strings can be shorter than 511 bytes, no problem, but if they are longer we need to cut them up.
 //
-// In each iteration of the while loop, we find one code in each of the unroll*8 strings, i.e. (8,16,24 or 32) for resp. unroll=1,2,3,4 
+// In each iteration of the while loop, we find one code in each of the unroll*8 strings, i.e. (8,16,24 or 32) for resp. unroll=1,2,3,4
 // unroll3 performs best on my hardware
 //
 // In the worst case, each final encoded string occupies 512KB bytes (512*1024; with 1024=512xexception, exception = 2 bytes).
-// - hence codeBase is a buffer of 512KB (needs 19 bits jobs), symbolBase of 256KB (needs 18 bits jobs). 
+// - hence codeBase is a buffer of 512KB (needs 19 bits jobs), symbolBase of 256KB (needs 18 bits jobs).
 //
 // 'jobX' controls the encoding of each string and is therefore a u64 with format [out:19][pos:9][end:18][cur:18] (low-to-high bits)
 // The field 'pos' tells which string we are processing (0..511). We need this info as strings will complete compressing out-of-order.
@@ -60,14 +60,14 @@ bool fsst_hasAVX512() { return false; }
 //
 // Apart from the coded strings, we return in a output[] array of size 'processed' the job values of the 'finished' strings.
 // In the following 'unfinished' slots (processed=finished+unfinished) we output the 'job' values of the unfinished strings.
-// 
+//
 // For the finished strings, we need [out:19] to see the compressed size and [pos:9] to see which string we refer to.
 // For the unfinished strings, we need all fields of 'job' to continue the compression with scalar code (see SIMD code in compressBatch).
 //
 // THIS IS A SEPARATE CODE FILE NOT BECAUSE OF MY LOVE FOR MODULARIZED CODE BUT BECAUSE IT ALLOWS TO COMPILE IT WITH DIFFERENT FLAGS
-// in particular, unrolling is crucial for gather/scatter performance, but requires registers. the #define all_* expressions however, 
-// will be detected to be constants by g++ -O2 and will be precomputed and placed into AVX512 registers - spoiling 9 of them. 
-// This reduces the effectiveness of unrolling, hence -O2 makes the loop perform worse than -O1 which skips this optimization. 
+// in particular, unrolling is crucial for gather/scatter performance, but requires registers. the #define all_* expressions however,
+// will be detected to be constants by g++ -O2 and will be precomputed and placed into AVX512 registers - spoiling 9 of them.
+// This reduces the effectiveness of unrolling, hence -O2 makes the loop perform worse than -O1 which skips this optimization.
 // Assembly inspection confirmed that 3-way unroll with -O1 avoids needless load/stores.
 
 size_t fsst_compressAVX512(SymbolTable &symbolTable, u8* codeBase, u8* symbolBase, SIMDjob *input, SIMDjob *output, size_t n, size_t unroll) {
@@ -88,7 +88,7 @@ size_t fsst_compressAVX512(SymbolTable &symbolTable, u8* codeBase, u8* symbolBas
 #define    all_FF         _mm512_srli_epi64(all_MASK, 56)
 
    SIMDjob *inputEnd = input+n;
-   assert(n >= unroll*8 && n <= 512); // should be close to 512 
+   assert(n >= unroll*8 && n <= 512); // should be close to 512
    __m512i job1, job2, job3, job4; // will contain current jobs, for each unroll 1,2,3,4
    __mmask8 loadmask1 = 255, loadmask2 = 255*(unroll>1), loadmask3 = 255*(unroll>2), loadmask4 = 255*(unroll>3); // 2b loaded new strings bitmask per unroll
    u32 delta1 = 8, delta2 = 8*(unroll>1), delta3 = 8*(unroll>2), delta4 = 8*(unroll>3); // #new loads this SIMD iteration per unroll
@@ -111,22 +111,22 @@ size_t fsst_compressAVX512(SymbolTable &symbolTable, u8* codeBase, u8* symbolBas
       }
    }
 
-   // flush the job states of the unfinished strings at the end of output[] 
+   // flush the job states of the unfinished strings at the end of output[]
    processed = n - (inputEnd - input);
    u32 unfinished = 0;
-   if (unroll > 1) { 
-      if (unroll > 2) { 
-         if (unroll > 3) { 
-            _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask4=~loadmask4, job4); 
+   if (unroll > 1) {
+      if (unroll > 2) {
+         if (unroll > 3) {
+            _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask4=~loadmask4, job4);
             unfinished += _mm_popcnt_u32((int) loadmask4);
          }
-         _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask3=~loadmask3, job3); 
+         _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask3=~loadmask3, job3);
          unfinished += _mm_popcnt_u32((int) loadmask3);
       }
-      _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask2=~loadmask2, job2); 
+      _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask2=~loadmask2, job2);
       unfinished += _mm_popcnt_u32((int) loadmask2);
    }
-   _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask1=~loadmask1, job1); 
+   _mm512_mask_compressstoreu_epi64(output+unfinished, loadmask1=~loadmask1, job1);
 #else
    (void) symbolTable;
    (void) codeBase;
