diff --git a/include/dwarfs/block_range.h b/include/dwarfs/block_range.h
index 3cb6446a..3860b760 100644
--- a/include/dwarfs/block_range.h
+++ b/include/dwarfs/block_range.h
@@ -31,6 +31,7 @@ class cached_block;
 
 class block_range {
  public:
+  block_range(void) = default;     // This is required for MSVC toolset v142 or less
   block_range(uint8_t const* data, size_t offset, size_t size);
   block_range(std::shared_ptr<cached_block const> block, size_t offset,
               size_t size);
diff --git a/include/dwarfs/compiler.h b/include/dwarfs/compiler.h
index 0a5edcdc..518b2291 100644
--- a/include/dwarfs/compiler.h
+++ b/include/dwarfs/compiler.h
@@ -37,6 +37,6 @@
 #endif
 #endif
 
-#if !defined(DWARFS_SANITIZE_THREAD) && defined(__x86_64__)
+#if !defined(DWARFS_SANITIZE_THREAD) && defined(FOLLY_HAVE_IFUNC) && defined(__x86_64__)
 #define DWARFS_MULTIVERSIONING 1
 #endif
diff --git a/include/dwarfs/console_writer.h b/include/dwarfs/console_writer.h
index 40c04c6b..d2ce0983 100644
--- a/include/dwarfs/console_writer.h
+++ b/include/dwarfs/console_writer.h
@@ -63,5 +63,6 @@ class console_writer : public stream_logger {
   bool const debug_progress_;
   bool writing_{false};
   speedometer<uint64_t> read_speed_;
+  bool fancy_{false};
 };
 } // namespace dwarfs
diff --git a/include/dwarfs/entry.h b/include/dwarfs/entry.h
index 60577abc..77166eec 100644
--- a/include/dwarfs/entry.h
+++ b/include/dwarfs/entry.h
@@ -29,6 +29,7 @@
 #include <optional>
 #include <string>
 #include <string_view>
+#include <unordered_map>
 #include <vector>
 
 #include <folly/small_vector.h>
diff --git a/include/dwarfs/fs_section.h b/include/dwarfs/fs_section.h
index 6b56b13b..3aa94019 100644
--- a/include/dwarfs/fs_section.h
+++ b/include/dwarfs/fs_section.h
@@ -33,6 +33,7 @@ class mmif;
 
 class fs_section {
  public:
+  fs_section(void) = default;	// This is required for MSVC toolset v142 or less
   fs_section(mmif const& mm, size_t offset, int version);
   fs_section(std::shared_ptr<mmif const> mm, section_type type, size_t offset,
              size_t size, int version);
diff --git a/include/dwarfs/lazy_value.h b/include/dwarfs/lazy_value.h
index 4526211e..c0397461 100644
--- a/include/dwarfs/lazy_value.h
+++ b/include/dwarfs/lazy_value.h
@@ -35,7 +35,11 @@ class lazy_value {
       : v_{f} {}
 
   T const& get() {
-    if (std::holds_alternative<function_type>(v_)) [[unlikely]] {
+    if (std::holds_alternative<function_type>(v_))
+#if __has_cpp_attribute(unlikely)
+    [[unlikely]]
+#endif
+    {
       v_ = std::get<function_type>(v_)();
     }
     return std::get<T>(v_);
diff --git a/include/dwarfs/offset_cache.h b/include/dwarfs/offset_cache.h
index eec8aa9d..fb76b167 100644
--- a/include/dwarfs/offset_cache.h
+++ b/include/dwarfs/offset_cache.h
@@ -136,7 +136,10 @@ class basic_offset_cache {
 
     void add_offset(chunk_index_type index, file_offset_type offset) {
       if (index < chunk_index_interval || index % chunk_index_interval != 0)
-          [[likely]] {
+#if __has_cpp_attribute(likely)
+          [[likely]]
+#endif
+      {
         return;
       }
 
diff --git a/include/dwarfs/program_options_helpers.h b/include/dwarfs/program_options_helpers.h
index eb44717a..f42997e1 100644
--- a/include/dwarfs/program_options_helpers.h
+++ b/include/dwarfs/program_options_helpers.h
@@ -21,13 +21,14 @@
 
 #pragma once
 
+#include <sstream>
 #include <boost/program_options.hpp>
 
 #include "dwarfs/types.h"
 
 namespace dwarfs {
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 template <typename T>
 auto po_sys_value(T* v) {
   return boost::program_options::wvalue<T>(v);
@@ -39,4 +40,15 @@ auto po_sys_value(T* v) {
 }
 #endif
 
+// https://stackoverflow.com/questions/45211248/boosts-is-any-of-causes-compile-warning
+inline std::vector<std::string> split(const std::string_view &str, char delim) {
+  std::vector<std::string> out;
+  std::stringstream ss((std::string(str)));
+  std::string item;
+  while (std::getline(ss, item, delim)) {
+      out.push_back(item);
+  }
+  return out;
+}
+
 } // namespace dwarfs
diff --git a/include/dwarfs/types.h b/include/dwarfs/types.h
index 37425354..ec1728d2 100644
--- a/include/dwarfs/types.h
+++ b/include/dwarfs/types.h
@@ -29,7 +29,7 @@ namespace dwarfs {
 
 using file_off_t = int64_t;
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 using sys_char = wchar_t;
 using sys_string = std::wstring;
 #define SYS_CERR std::wcerr
diff --git a/include/dwarfs/util.h b/include/dwarfs/util.h
index 2934216b..5dc9ec9e 100644
--- a/include/dwarfs/util.h
+++ b/include/dwarfs/util.h
@@ -43,6 +43,14 @@ inline std::string u8string_to_string(std::u8string const& in) {
   return std::string(reinterpret_cast<char const*>(in.data()), in.size());
 }
 
+// std::string u8string() const;  (since C++17)  (until C++20)
+// std::u8string u8string() const; (since C++20)
+
+inline std::string u8string_to_string(std::string const& in) {
+  return in;
+}
+
+
 std::string sys_string_to_string(sys_string const& in);
 
 size_t utf8_display_width(char const* p, size_t len);
diff --git a/src/dwarfs/cached_block.cpp b/src/dwarfs/cached_block.cpp
index faa3febe..c550db48 100644
--- a/src/dwarfs/cached_block.cpp
+++ b/src/dwarfs/cached_block.cpp
@@ -93,7 +93,7 @@ class cached_block_ final : public cached_block {
 
   bool any_pages_swapped_out(std::vector<uint8_t>& tmp
                              [[maybe_unused]]) const override {
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__MACH__)  // tebako patched - we do not expect to have fuse runtime on Mach
     auto page_size = ::sysconf(_SC_PAGESIZE);
     tmp.resize((data_.size() + page_size - 1) / page_size);
     if (::mincore(const_cast<uint8_t*>(data_.data()), data_.size(),
diff --git a/src/dwarfs/checksum.cpp b/src/dwarfs/checksum.cpp
index 1876368a..20cd2e78 100644
--- a/src/dwarfs/checksum.cpp
+++ b/src/dwarfs/checksum.cpp
@@ -36,6 +36,8 @@
 
 namespace dwarfs {
 
+using namespace std::literals::string_view_literals;
+
 namespace {
 
 std::unordered_set<std::string> supported_algorithms{
@@ -43,6 +45,11 @@ std::unordered_set<std::string> supported_algorithms{
     "xxh3-128",
 };
 
+constexpr std::array unsupported_algorithms{
+    "shake128"sv,
+    "shake256"sv,
+};
+
 class checksum_evp : public checksum::impl {
  public:
   explicit checksum_evp(::EVP_MD const* evp)
@@ -76,7 +83,8 @@ class checksum_evp : public checksum::impl {
     std::vector<std::string> available;
     ::EVP_MD_do_all(
         [](const ::EVP_MD*, const char* from, const char* to, void* vec) {
-          if (!to) {
+          if (!to && std::ranges::find(unsupported_algorithms, from) ==
+                         unsupported_algorithms.end()) {
             reinterpret_cast<std::vector<std::string>*>(vec)->emplace_back(
                 from);
           }
diff --git a/src/dwarfs/console_writer.cpp b/src/dwarfs/console_writer.cpp
index 59410139..f52adb25 100644
--- a/src/dwarfs/console_writer.cpp
+++ b/src/dwarfs/console_writer.cpp
@@ -76,10 +76,11 @@ console_writer::console_writer(std::ostream& os, progress_mode pg_mode,
     , get_term_width_(get_term_width)
     , mode_(mode)
     , debug_progress_(getenv_is_enabled("DWARFS_DEBUG_PROGRESS"))
-    , read_speed_{std::chrono::seconds(5)} {}
+    , read_speed_{std::chrono::seconds(5)}
+    , fancy_(pg_mode_ == UNICODE || pg_mode_ == ASCII) {}
 
 void console_writer::rewind() {
-  if (!statebuf_.empty()) {
+  if (!statebuf_.empty() && fancy_) {
     int lines = 0;
 
     switch (mode_) {
@@ -104,13 +105,13 @@ void console_writer::rewind() {
 void console_writer::preamble() { rewind(); }
 
 void console_writer::postamble() {
-  if (pg_mode_ == UNICODE || pg_mode_ == ASCII) {
+  if (fancy_) {
     log_stream() << statebuf_;
   }
 }
 
 std::string_view console_writer::get_newline() const {
-  return pg_mode_ != NONE ? "\x1b[K\n" : "\n";
+  return fancy_ ? "\x1b[K\n" : "\n";
 }
 
 void console_writer::update(const progress& p, bool last) {
@@ -123,10 +124,8 @@ void console_writer::update(const progress& p, bool last) {
   std::ostringstream oss;
   lazy_value width(get_term_width_);
 
-  bool fancy = pg_mode_ == ASCII || pg_mode_ == UNICODE;
-
-  if (last || fancy) {
-    if (fancy) {
+  if (last || fancy_) {
+    if (fancy_) {
       for (size_t i = 0; i < width.get(); ++i) {
         oss << (pg_mode_ == UNICODE ? "⎯" : "-");
       }
@@ -146,13 +145,11 @@ void console_writer::update(const progress& p, bool last) {
         }
       }
 
-      if (fancy) {
+      if (fancy_) {
         oss << terminal_colored(p.status(width.get()), termcolor::BOLD_CYAN,
                                 log_is_colored())
             << newline;
-      }
 
-      {
         auto cur_size = p.current_size.load();
         double cur_offs = std::min(p.current_offset.load(), cur_size);
         double cur_frac = cur_size > 0 ? cur_offs / cur_size : 0.0;
@@ -233,7 +230,7 @@ void console_writer::update(const progress& p, bool last) {
     frac_ = frac;
   }
 
-  if (pg_mode_ == SIMPLE) {
+  if (!fancy_) {
     std::string tmp =
         fmt::format(" ==> {0:.0f}% done, {1} blocks/{2} written", 100 * frac_,
                     p.blocks_written.load(), size_with_unit(p.compressed_size));
diff --git a/src/dwarfs/entry.cpp b/src/dwarfs/entry.cpp
index 7d310f0d..48741ef5 100644
--- a/src/dwarfs/entry.cpp
+++ b/src/dwarfs/entry.cpp
@@ -65,10 +65,17 @@ std::u8string entry::u8name() const { return string_to_u8string(name_); }
 
 std::filesystem::path entry::fs_path() const {
   if (auto parent = parent_.lock()) {
+#ifdef U8STRING_AND_PATH_OK
     return parent->fs_path() / u8name();
+#else
+    return parent->fs_path() / name_;
+#endif
   }
-
+#ifdef U8STRING_AND_PATH_OK
   return std::filesystem::path(u8name());
+#else
+  return std::filesystem::path(name_);
+#endif
 }
 
 std::string entry::path_as_string() const {
diff --git a/src/dwarfs/file_stat.cpp b/src/dwarfs/file_stat.cpp
index d9d35f26..d8d9320a 100644
--- a/src/dwarfs/file_stat.cpp
+++ b/src/dwarfs/file_stat.cpp
@@ -165,9 +165,15 @@ file_stat make_file_stat(fs::path const& path) {
   rv.size = st.st_size;
   rv.blksize = st.st_blksize;
   rv.blocks = st.st_blocks;
-  rv.atime = st.st_atim.tv_sec;
-  rv.mtime = st.st_mtim.tv_sec;
-  rv.ctime = st.st_ctim.tv_sec;
+
+  rv.atime = st.st_atime;
+  rv.mtime = st.st_mtime;
+  rv.ctime = st.st_ctime;
+
+//  Tebako -- using legacy syntax above to comply with MacOS definitions (~ FreeBSD)
+//  rv.atime = st.st_atim.tv_sec;
+//  rv.mtime = st.st_mtim.tv_sec;
+//  rv.ctime = st.st_ctim.tv_sec;
 
   return rv;
 }
diff --git a/src/dwarfs/file_type.cpp b/src/dwarfs/file_type.cpp
index e77e2d73..50e7abcc 100644
--- a/src/dwarfs/file_type.cpp
+++ b/src/dwarfs/file_type.cpp
@@ -77,7 +77,7 @@ uint16_t file_status_to_mode(std::filesystem::file_status status) {
     ft = posix_file_type::block;
     break;
   case fs::file_type::directory:
-#ifdef _WIN32
+#ifdef _MSC_VER
   case fs::file_type::junction:
 #endif
     ft = posix_file_type::directory;
diff --git a/src/dwarfs/filesystem_v2.cpp b/src/dwarfs/filesystem_v2.cpp
index e5bc88b0..102b0ffa 100644
--- a/src/dwarfs/filesystem_v2.cpp
+++ b/src/dwarfs/filesystem_v2.cpp
@@ -73,7 +73,12 @@ class filesystem_parser {
       auto ss = mm.span<char>(start);
       auto it = std::search(
           ss.begin(), ss.end(),
-          std::boyer_moore_horspool_searcher(magic.begin(), magic.end()));
+#if __cpp_lib_boyer_moore_searcher
+          std::boyer_moore_horspool_searcher
+#else
+          std::default_searcher
+#endif
+          (magic.begin(), magic.end()));
 
       if (it == ss.end()) {
         break;
diff --git a/src/dwarfs/metadata_types.cpp b/src/dwarfs/metadata_types.cpp
index c0751c7c..d387422e 100644
--- a/src/dwarfs/metadata_types.cpp
+++ b/src/dwarfs/metadata_types.cpp
@@ -701,7 +701,11 @@ void dir_entry_view::append_to(std::filesystem::path& p) const {
     }
   }
   if (!is_root()) {
+#ifdef U8STRING_AND_PATH_OK
     p /= string_to_u8string(name());
+#else
+    p /= name();
+#endif
   }
 }
 
diff --git a/src/dwarfs/metadata_v2.cpp b/src/dwarfs/metadata_v2.cpp
index 052eb594..ed40fa1a 100644
--- a/src/dwarfs/metadata_v2.cpp
+++ b/src/dwarfs/metadata_v2.cpp
@@ -847,7 +847,7 @@ void metadata_<LoggerPolicy>::dump(
     time_t tp = *ts;
     std::string str(32, '\0');
     str.resize(
-        std::strftime(str.data(), str.size(), "%F %T", std::localtime(&tp)));
+        std::strftime(str.data(), str.size(), "%Y-%m-%d %H:%M:%S", std::localtime(&tp)));
     os << "created on: " << str << "\n";
   }
 
diff --git a/src/dwarfs/option_map.cpp b/src/dwarfs/option_map.cpp
index cc0090eb..68a54a6b 100644
--- a/src/dwarfs/option_map.cpp
+++ b/src/dwarfs/option_map.cpp
@@ -28,18 +28,26 @@
 
 #include "dwarfs/error.h"
 #include "dwarfs/option_map.h"
+#include "dwarfs/program_options_helpers.h"
 
 namespace dwarfs {
 
 option_map::option_map(const std::string_view spec) {
-  std::vector<std::string_view> arg;
-  boost::split(arg, spec, boost::is_any_of(":"));
+
+  auto arg = split(spec,':');
+
+//  https://stackoverflow.com/questions/45211248/boosts-is-any-of-causes-compile-warning
+//  std::vector<std::string_view> arg;
+//  boost::split(arg, spec, boost::is_any_of(std::string(":")));
 
   choice_ = arg[0];
 
   for (size_t i = 1; i < arg.size(); ++i) {
-    std::vector<std::string> kv;
-    boost::split(kv, arg[i], boost::is_any_of("="));
+      auto kv = split(arg[i], '=');
+
+//    https://stackoverflow.com/questions/45211248/boosts-is-any-of-causes-compile-warning
+//    std::vector<std::string> kv;
+//    boost::split(kv, arg[i], boost::is_any_of(std::string("=")));
 
     if (kv.size() > 2) {
       DWARFS_THROW(runtime_error,
diff --git a/src/dwarfs/safe_main.cpp b/src/dwarfs/safe_main.cpp
index 0c47ac20..5959a1a8 100644
--- a/src/dwarfs/safe_main.cpp
+++ b/src/dwarfs/safe_main.cpp
@@ -38,14 +38,17 @@ int safe_main(std::function<int(void)> fn) {
     folly::symbolizer::installFatalSignalHandler();
 #endif
     try {
-#ifdef _WIN32
+#if defined(_MSC_VER)
       char const* locale = "en_US.utf8";
+#elif defined(__MINGW32__)
+     //Only "C" and "POSIX" locales are supported under MinGW port of GCC and libstdc++.
+      char const* locale = "C";
 #else
       char const* locale = "";
 #endif
       std::locale::global(std::locale(locale));
       if (!std::setlocale(LC_ALL, locale)) {
-        std::cerr << "warning: setlocale(LC_ALL, \"\") failed\n";
+        std::cerr << "warning: setlocale(LC_ALL, \"" << locale << "\") failed\n";
       }
     } catch (std::exception const& e) {
       std::cerr << "warning: failed to set user default locale: " << e.what()
diff --git a/src/dwarfs/util.cpp b/src/dwarfs/util.cpp
index c6eab260..f4692069 100644
--- a/src/dwarfs/util.cpp
+++ b/src/dwarfs/util.cpp
@@ -24,7 +24,11 @@
 #include <charconv>
 #include <cstdlib>
 
+#ifdef _MSC_VER
 #include <utf8cpp/utf8.h>
+#else
+#include <utf8.h>
+#endif
 
 #include <folly/String.h>
 
@@ -122,7 +126,7 @@ std::chrono::milliseconds parse_time_with_unit(std::string const& str) {
 }
 
 std::string sys_string_to_string(sys_string const& in) {
-#ifdef _WIN32
+#ifdef _MSC_VER
   std::u16string tmp(in.size(), 0);
   std::transform(in.begin(), in.end(), tmp.begin(),
                  [](sys_char c) { return static_cast<char16_t>(c); });
diff --git a/src/dwarfs/worker_group.cpp b/src/dwarfs/worker_group.cpp
index 3c04b630..e9e6b849 100644
--- a/src/dwarfs/worker_group.cpp
+++ b/src/dwarfs/worker_group.cpp
@@ -31,6 +31,20 @@
 #include <type_traits>
 #include <vector>
 
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
+#if __MACH__
+#include <mach/mach.h>
+#include <mach/mach_time.h>
+#include <mach/thread_info.h>
+#include <mach/mach_vm.h>
+#include <mach/mach_init.h>
+#include <mach/mach_port.h>
+#include <mach/thread_act.h>
+#endif
+
 #include <folly/Conv.h>
 #include <folly/portability/PThread.h>
 #include <folly/portability/Windows.h>
@@ -45,7 +59,7 @@ namespace dwarfs {
 
 namespace {
 
-#ifdef _WIN32
+#ifdef _MSC_VER
 
 double get_thread_cpu_time(std::thread const& t) {
   static_assert(sizeof(std::thread::id) == sizeof(DWORD),
@@ -77,12 +91,35 @@ pthread_t std_to_pthread_id(std::thread::id tid) {
 }
 
 double get_thread_cpu_time(std::thread const& t) {
+#if __MACH__
+//  https://www.programcreek.com/cpp/?CodeExample=thread+cpu+usage
+      mach_msg_type_number_t count;
+      thread_basic_info_data_t info;
+      count = THREAD_BASIC_INFO_COUNT;
+      mach_port_t thread = pthread_mach_thread_np(std_to_pthread_id(t.get_id()));
+      if (::thread_info(thread, THREAD_BASIC_INFO, (thread_info_t)&info, &count) == KERN_SUCCESS) {
+        return (info.user_time.seconds + info.user_time.microseconds * 1e-6) +
+               (info.system_time.seconds + info.system_time.microseconds * 1e-6);
+      }
+#elif __MINGW32__
+      FILETIME CreationTime, ExitTime, KernelTime, UserTime;
+// pthread_gethandle is MINGW private extension
+      double w = 0.0;
+      HANDLE hThread = pthread_gethandle(std_to_pthread_id(t.get_id()));
+      BOOL r = GetThreadTimes(hThread, &CreationTime, &ExitTime, &KernelTime, &UserTime);
+      if (r) {
+        w = UserTime.dwLowDateTime * 1e-7 + KernelTime.dwLowDateTime * 1e-7;
+      }
+      return w;
+// GetThreadTimes is not really reliable
+#else
   ::clockid_t cid;
   struct ::timespec ts;
   if (::pthread_getcpuclockid(std_to_pthread_id(t.get_id()), &cid) == 0 &&
       ::clock_gettime(cid, &ts) == 0) {
     return ts.tv_sec + 1e-9 * ts.tv_nsec;
   }
+#endif
   throw std::runtime_error("get_thread_cpu_time");
 }
 
